Create Todo-list

VolkovaKaterina opened the pull request.

VolkovaKaterina committed:
Create Todo-list

OlehRovenskyi commented:
Нужно использовать что то одно либо npm или yarn у тебя два разных фала lock, один из них нужно убрать.

OlehRovenskyi commented on db/db.json:
```diff
@@ -0,0 +1,24 @@
+    {
+      "task": "go to shop",
+      "completed": false,
+      "id": "2e9f9c"
```
id в данном использовании лучше делать в одном формате.

OlehRovenskyi commented on src/api/todoApi.js:
```diff
@@ -0,0 +1,3 @@
+import axios from 'axios';
+
+export default axios.create({ baseURL: 'http://localhost:3004' });
```
Нужно использовать методы post, get, put, patch ....

OlehRovenskyi replied:
Вернее вынести сюда сами запросы, отделить имплементацию api call из компонента.

OlehRovenskyi commented:
В README.md нужно добавить секцию How to run с описание как запустить проект.

OlehRovenskyi commented on src/componets/App.js:
```diff
@@ -0,0 +1,111 @@
+  const [todos, setTodos] = useState([]);
+  const [filterStatus, setFilterStatus] = useState('all');
+  const [filterTodos, setFilterTodos] = useState([]);
+  const uniqueId = uuid();
```
Id не нужно генерировать, его генерация ответственность back-end

OlehRovenskyi commented on src/componets/App.js:
```diff
@@ -0,0 +1,111 @@
+        const newTodo = await todoApi.post('/todos', {
+          task: input,
+          completed: false,
+          id: smallId,
```
При создании не нужно отправлять id

OlehRovenskyi commented on src/componets/App.js:
```diff
@@ -0,0 +1,111 @@
+  const uniqueId = uuid();
+  const smallId = uniqueId.slice(0, 6);
+
+  useEffect(() => {
```
почитай о том как делать async useEffect. Не нужно создавать отдельную функцию и затем ее вызывать. Выполнения запроса на получение todos должно быть проще. Саму реализацию запроса нужно вынести. Посмотри комментарий выше.

OlehRovenskyi commented on src/componets/App.js:
```diff
@@ -0,0 +1,111 @@
+          setTodos([...todos, newTodo.data]);
+        }
+      } catch (err) {
+        console.log(err);
```
Вместо log лучше error использовать.

OlehRovenskyi commented on src/componets/App.js:
```diff
@@ -0,0 +1,111 @@
+      console.log(err);
+    }
+  };
+  const EditTodo = async (value, todo) => {
```
Почему EditTodo с большой буквы начинается?

OlehRovenskyi commented on src/componets/Todo.js:
```diff
@@ -0,0 +1,52 @@
+import { BsTrash, BsSave2 } from 'react-icons/bs';
+import { AiOutlineCheck, AiFillEdit } from 'react-icons/ai';
+
+const Todo = ({
```
Лучше переименовать, что бы имя больше отображало суть того за что отвечает данный компонент.

OlegRovenskyi commented on src/componets/TodoList.js:
```diff
@@ -0,0 +1,24 @@
+}) => (
+  <div className="todo-container">
+    <ul className="todo-list">
+      {filterTodos.map(((todo) => (
```
НУжно применить механизм что бы все todo не перерисовывались когда отредактировали одну todo или добавили новую todo

OlehRovenskyi commented on src/componets/Todo.js:
```diff
@@ -0,0 +1,52 @@
+  };
+
+  return (
+    <div className="todo">
```
li не должна быть вложена в div, для пустого враппера есть специальный синтаксис <></>

OlehRovenskyi commented on src/componets/TodoList.js:
```diff
@@ -0,0 +1,24 @@
+          task={todo.task}
+          completeTodo={completeTodo}
+          deleteTodo={deleteTodo}
+          EditTodo={EditTodo}
```
почему EditTodo с большой буквы?

VolkovaKaterina committed:
moved api requests to the TodoService file
change console log to the console.error
delete id generation
Todo component rename to TodoItem
add installation to the Read me file

oleh-rovenskyi-nortal commented on src/componets/App.js:
```diff
@@ -0,0 +1,106 @@
+  useEffect(() => { setHeight(document.documentElement.scrollHeight); });
+
+  useEffect(() => {
+    (async () => {
```
Вместо самовызывающейся функции можно использовать useEffect(async() => {.....

oleh-rovenskyi-nortal commented on src/componets/TodoItem.js:
```diff
@@ -0,0 +1,56 @@
+          </form>
+        ) : `${editTask}`}
+      </li>
+      <div className="todo-buttons__container">
```
Желательно что бы кнопки для работы с item были в одну строку с input в приложении, сейчас кнопки переносятся на новую строку.

VolkovaKaterina committed:
fix: useEffect in the App component
change: the buttons in the PostItem component to one line with item

OlehRovenskyi commented on src/componets/Todoform.css:
```diff
@@ -0,0 +1,149 @@
+* {
```
Instead css better to use scss or css4
